src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.cpp

```diff
@@ -568,6 +579,89 @@ void SolidMechanicsLagrangeContact::assembleContact( DomainPartition & domain,
   } );
 }
 
+void SolidMechanicsLagrangeContact::
+  assembleForceResidualPressureContribution( MeshLevel const & mesh,
+                                             arrayView1d< string const > const & regionNames,
+                                             DofManager const & dofManager,
+                                             CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                             arrayView1d< real64 > const & localRhs )
+{
+  GEOS_MARK_FUNCTION;
+
+  FaceManager const & faceManager = mesh.getFaceManager();
+  NodeManager const & nodeManager = mesh.getNodeManager();
+  ElementRegionManager const & elemManager = mesh.getElemManager();
+
+  ArrayOfArraysView< localIndex const > const & faceToNodeMap = faceManager.nodeList().toViewConst();
+  arrayView2d< real64 const > const & faceNormal = faceManager.faceNormal();
+
+  string const & dispDofKey = dofManager.getKey( solidMechanics::totalDisplacement::key() );
+
+  arrayView1d< globalIndex const > const &
+  dispDofNumber = nodeManager.getReference< globalIndex_array >( dispDofKey );
+  globalIndex const rankOffset = dofManager.rankOffset();
+
+  // Get the coordinates for all nodes
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const & nodePosition = nodeManager.referencePosition();
+
+  elemManager.forElementSubRegions< FaceElementSubRegion >( regionNames,
+                                                            [&]( localIndex const,
+                                                                 FaceElementSubRegion const & subRegion )
+  {
+    arrayView1d< real64 const > const & pressure = subRegion.getReference< array1d< real64 > >( flow::pressure::key() );
+    ArrayOfArraysView< localIndex const > const & elemsToFaces = subRegion.faceList().toViewConst();
+
+    forAll< serialPolicy >( subRegion.size(), [=]( localIndex const kfe )
```

Comment:
I wonder why this needs to be a serialPolicy.

Reply:
this comes from here https://github.com/GEOS-DEV/GEOS/blob/d814b2213c1d15a988ff56581ecd64a25bea9dd2/src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsConformingFractures.cpp#L483
the whole new assembly function (no derivatives) is a tailored version of old assembly function (with derivatives)

Reply:
yeah, that one has issues too. It all started from the initial implementation. Can you create an issue and assign me to it to clean these up. Some things are easy to do. For example there is a localArray that can easily be turned into a c-array. For all others we should be using stackArrays. I think that if we fix those things we should be able to run these kernels on the gpus.

Reply:
created here https://github.com/GEOS-DEV/GEOS/issues/3036